string name
bool available
float32 idleness